app/src/main/java/com/teamagam/gimelgimel/app/map/esri/EsriGGMapView.java

```diff
@@ -148,12 +165,35 @@ private boolean isBaseLayerLoaded(Object o, STATUS status) {
     }
 
     private void onBasemapLoaded() {
-        setExtentOverIsrael();
+        setExtent();
         addDynamicGraphicLayer();
         notifyMapReady();
         setOnStatusChangedListener(null);
     }
 
+    private void setExtent() {
```

Comment:
You can delete this method and add an else clause to restoreState, which calls setExtentOverIsrael.
As for now it's kinda duplicate (restoreState and setExtent shares code)...